### ISSUE TITLE
Build fails with SSR turned off

### ISSUE THREAD

**Issue by francisfontoura** [nuxt/framework#886]
###  Environment

- Operating System: `Linux`
- Node Version:     `v14.18.0`
- Nuxt Version:     `2.16.0-27226092.034b9901`
- Package Manager:  `Yarn`
- Bundler:          `Webpack`
- User Config:      `ssr`, `buildModules`
- Runtime Modules:  `-`
- Build Modules:    `@nuxt/bridge@3.0.0-27234503.d5127e9`

### Describe the bug and expected behavior

`nuxi build` fails with `ssr: false`. Output:
```
start Building server...                                                                                                                                               00:26:02

 ERROR  Rollup error: Could not load .nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '.nuxt/dist/server/server.mjs'


 ERROR  Nuxt Build Error: Could not load .nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '.nuxt/dist/server/server.mjs'
```

### Steps to reproduce

1. Set `ssr: false` in `nuxt.config.ts`.
2. Run (`npx`) `nuxi build` in shell.

### Additional context

My project doesn't need SSR, but needs "runtime config" with environment variables injected by Node.js.

**Comment by ackushiw-flutterwave** [comment]
As a workaround, I got it building successfully by updating the nuxt config:  

```js

bridge: {
  nitro: false,
},
```

**Comment by namcodes** [comment]
same problem

**Comment by victorberland** [comment]
Considering this issue has been open for more than a month now without any progress on the matter, does anyone maybe know of a temporary fix that may work?

**Comment by AllInOneYT** [comment]
> Considering this issue has been open for more than a month now without any progress on the matter, does anyone maybe know of a temporary fix that may work?

you could turn off nitro temporary

* nuxt.config.ts
```ts
defineNuxtConfig({
   bridge: {
       nitro: false
   }
})
```

* package.json scripts are the same
```json
{
    "scripts": {
        "build": "nuxi build",
        "start": "nuxt start"
    }
}
```

**Comment by vyvian1712** [comment]
I also have the same error :(
How to fix it???

**Comment by piscis** [comment]
@rodrigogs with `3.0.0-27313139.1c88580` it's still broken. Building my project with ssr false leads to this:

```
$ nuxi build
Nuxt CLI v3.0.0-27313139.1c88580                                                         16:35:25
ℹ Vite warmed up in 3822ms                                                               16:35:30
ℹ Client built in 3915ms                                                                 16:35:30
ℹ Nitro preset is server                                                                 16:35:30
ℹ Cleaning up .output                                                                    16:35:30
start Generating public...                                                               16:35:30
✔ Generated public .output/public                                                        16:35:30
start Building server...                                                                 16:35:30

 ERROR  Rollup error: Could not load /Users/alex/nuxt-quasar-boilerplate/.nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '/Users/alex/nuxt-quasar-boilerplate/.nuxt/dist/server/server.mjs'


 ERROR  Could not load /Users/alex/nuxt-quasar-boilerplate/.nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '/Users/alex/nuxt-quasar-boilerplate/.nuxt/dist/server/server.mjs

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
**Repo:** https://github.com/piscis/nuxt-quasar-boilerplate/tree/example/ssr-fix-testing

**Comment by Aurelius333** [comment]
Yeah, my pull request (#2191) doesn't seem to have actually fixed this issue. It's working in Nuxt's `playground` directory, but for some reason, if you actually create a project, Rollup still thinks it needs to import `#build/dist/server/server.mjs`.

**Comment by rodrigogs** [comment]
I think this issue should be reopened then.

**Comment by m0uka** [comment]
Same issue here, please reopen

**Comment by pi0** [comment]
I just wanted to confirm if this workaround works or not. We might need to generate stub server.mjs for spa mode.

**Comment by rodrigogs** [comment]
Thank you @pi0, I can help testing it if you need. Just let me know :)

**Comment by yeliulee** [comment]
any update on this

**Comment by terion-name** [comment]
Sema here. Build works with `ssr: false` and `target: static` only, but static output is strange and can't figure out how to properly serve it

**Comment by rodrigogs** [comment]
> Same here. Build works with `ssr: false` and `target: static` only, but static output is strange and can't figure out how to properly serve it

It's not working atm

**Comment by joel-wenzel** [comment]
@pi0 Appreciate all the work you guys are doing. Just curious if you think spa output generation will get added during the beta or not until the official release of nuxt 3

**Comment by baixiaoyu2997** [comment]
> > Considering this issue has been open for more than a month now without any progress on the matter, does anyone maybe know of a temporary fix that may work?
> 
> you could turn off nitro temporary
> 
> * nuxt.config.ts
> 
> ```ts
> defineNuxtConfig({
>    bridge: {
>        nitro: false
>    }
> })
> ```

not work for me

**Comment by rodrigogs** [comment]
> > > Considering this issue has been open for more than a month now without any progress on the matter, does anyone maybe know of a temporary fix that may work?
> > 
> > 
> > you could turn off nitro temporary
> > 
> > * nuxt.config.ts
> > 
> > ```ts
> > defineNuxtConfig({
> >    bridge: {
> >        nitro: false
> >    }
> > })
> > ```
> 
> not work for me

It works for me, what is the error when you disable nitro?

**Comment by yeliulee** [comment]
> > > > Considering this issue has been open for more than a month now without any progress on the matter, does anyone maybe know of a temporary fix that may work?
> > > 
> > > 
> > > you could turn off nitro temporary
> > > 
> > > * nuxt.config.ts
> > > 
> > > ```ts
> > > defineNuxtConfig({
> > >    bridge: {
> > >        nitro: false
> > >    }
> > > })
> > > ```
> > 
> > not work for me
> 
> It works for me, what is the error when you disable nitro?

but it doesn't work with nuxt3

**Comment by joel-wenzel** [comment]
@rodrigogs I actually dont get an error but no ```.output``` or ```.dist``` content is generated. It seems like it might be generating under ```.nuxt/dist``` but that folder has both client and server folders and the client folder contains no index.html

This is on the latest version of @nuxt/bridge ```3.0.0-27338323.1e98259```

**Comment by AllInOneYT** [comment]
> @rodrigogs I actually dont get an error but no `.output` or `.dist` content is generated. It seems like it might be generating under `.nuxt/dist` but that folder has both client and server folders and the client folder contains no index.html
> 
> This is on the latest version of @nuxt/bridge `3.0.0-27338323.1e98259`

to build your project first run `nuxi build` then `nuxi generate`

> Same here. Build works with `ssr: false` and `target: static` only, but static output is strange and can't figure out how to properly serve it

and serve it with `nuxt start`

**Comment by baixiaoyu2997** [comment]
@rodrigogs it's works, i made a mistake, `generate ` dir should be `/dist`，but i view in to `/.nuxt/dist/` .
and thanks @AllInOneYT .
I write the complete steps below, hoping to help others.

1. config:
```js
// nuxt.config.js
const config = defineNuxtConfig({
  ssr:false,
  target:"static",
  bridge: {
      nitro:false,
   },
})
```

2. command:
```sh
nuxi generate
```
3. check dir `/dist/`

my env:
```
- Operating System: `Darwin`
- Node Version:     `v16.13.1`
- Nuxt Version:     `2.16.0-27355697.777a4b7f`
- Package Manager:  `yarn@1.22.17`
- Bundler:          `Webpack`
- User Config:      `build`, `bridge`, `buildModules`, `meta`, `css`, `plugins`, `components`, `ssr`, `target`, `modules`
- Runtime Modules:  `-`
- Build Modules:    `@nuxt/bridge@3.0.0-27338323.1e98259`, `@nuxt/bridge@3.0.0-27338323.1e98259`, `@nuxt/bridge@3.0.0-27338323.1e98259`, `@nuxt/bridge@3.0.0-27338323.1e98259`, `@vueuse/core/nuxt@7.5.1`, `@pinia/nuxt@0.1.5`, `@nuxtjs/eslint-module@3.0.2`, `@nuxtjs/stylelint-module@4.1.0`
```

**Comment by Chris2011** [comment]
The workaround with the bridge just works if you use Nuxt v2, not Nuxt v3. And `nuxt generate` also not working, it is still WIP.

**Comment by blazmrak** [comment]
Guys, am I stupid or is there no reason for this [line](https://github.com/nuxt/framework/blob/59e5cfea39e7854b1c89bc929a3208c8bfe220c4/packages/vite/src/vite.ts#L110)? Uncommenting it builds the server and running it works, the only issue I have with it is that I lose the css from app.vue, which does not seem to get requested.

**Comment by blazmrak** [comment]
The issue with css is fixed by setting ssr: false [here](https://github.com/nuxt/framework/blob/59e5cfea39e7854b1c89bc929a3208c8bfe220c4/packages/bridge/src/vite/server.ts#L55). Which basically means two minor changes I think (disclaimer: I have no idea what get's impacted): remove the `if(ssr) then buildServer` and have the ssr in config be set to `config.ssr ?? true`. 

Another thing: is the mergeConfig the wrong way around? Right now the ctx.config get's overriden by the (at least from what I understand) default config. Vite API is defined as `mergeConfig(oldConfig, newConfig)` from what I have seen.

**Comment by madsh93** [comment]
To understand it correctly. It is not possible to deploy SPAs with Nuxt 3 right now?
With SPA I mean SSR: false.

**Comment by MartinX3** [comment]
I use `target: static` and `ssr: true` as a workaround to deploy my static site.
Of course it doubles the deployment time, since it compiles a static and a server version of the website.

**Comment by blazmrak** [comment]
@madsh93 no it isn't UNLESS you go and edit 4 lines of code in node_modules -> check my above comments for the lines and files, but I have no idea if it actually works as it should, because I didn't get to test it much.

**Comment by madsh93** [comment]
@MartinX3 I just tested this, and my every component shows double. I think it may be because of using Teleport which isn't that SSR friendly yet, as I understand it.

@blazmrak Thanks. I'll give it it a try and come back with a response.

**Comment by blazmrak** [comment]
@madsh93 I have just remembered that I have linked the line packages/bridge/vite/server.ts, but it is actually packages/vite/server.ts in my second comment. I have edited the comment.

**Comment by MartinX3** [comment]
@madsh93 Sorry, I forgot to mention that I use the Nuxt 2 Bridge and not Nuxt 3.
There it works if I only deploy `.output/public/` and ignore `.output/server` (it was `dist/` without the bridge and SSR in Nuxt 2)

**Comment by rodrigogs** [comment]
If `3.0.0-27367970.e0fec80` contains this fix, then it still not working :/

**Comment by blazmrak** [comment]
@rodrigogs Nuxt 3 or Bridge? I did not touch Bridge, which still has ssr: true on build.

**Comment by rodrigogs** [comment]
> @rodrigogs Nuxt 3 or Bridge? I did not touch Bridge, which still has ssr: true in server.ts

Bridge

**Comment by blazmrak** [comment]
@rodrigogs you can change node_module/.../bridge/vite/server.ts ssr:true to ssr:false and give it a try :) @pi0 Would it make sense to replicate `ssr: ctx.nuxt.options.ssr ?? true` in Bridge? I am fresh to Nuxt 3 and have no idea what needs to be replicated in Bridge and what doesn't.

**Comment by blazmrak** [comment]
@rodrigogs it isn't - [at least not this one](https://github.com/nuxt/framework/blob/74be5a74166c18a1a9b81718b84230e80a033541/packages/bridge/src/vite/server.ts#L55)

**Comment by rodrigogs** [comment]
What?

I have `ssr: false` in `nuxt.config.js`

**Comment by blazmrak** [comment]
@rodrigogs #2708 should fix your issue. The problem is that bridge is not reading the ssr value from config at the moment, so your ssr: false from nuxt.config.js is ignored :D

**Comment by rodrigogs** [comment]
> @rodrigogs #2708 should fix your issue. The problem is that bridge is not reading the ssr value from config at the moment, so your ssr: false from nuxt.config.js is ignored :D

How can I set it then?

**Comment by MartinX3** [comment]
@blazmrak Are you sure if it is ignored with bridge?
If I set it to false, the build fails, but if I set it to true, it succeeds.

**Comment by MartinX3** [comment]
@blazmrak 
Yes, with bridge and `SSR: false` it only builds the client and fails, while with `SSR: true` it builds the client and server, both succesfully.
The error is the same as above

```
[error] Rollup error: Could not load node_modules/.cache/nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open 'node_modules/.cache/nuxt/dist/server/server.mjs'
 [fatal] Could not load node_modules/.cache/nuxt/dist/server/server.mjs (imported by node_modules/@nuxt/nitro/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open 'node_modules/.cache/nuxt/dist/server/server.mjs'
```

**Comment by blazmrak** [comment]
@MartinX3 what happens if you put vite: true in the config?

**Comment by blazmrak** [comment]
@pi0 I think the same thing is going on with webpack where server gets created only if ssr is enabled. Is this some legacy behavior or was the case for an SPA and API in the same project not planned?

**Comment by rodrigogs** [comment]
> @rodrigogs I have just made the PR for bridge. You have to change the file in node_modules. For nuxt3 it is node_modules/@nuxt/vite-builder/dist/index.mjs @MartinX3 I am sure - I have linked the line where it is ssr: true instead of taking into account the config. What might be causing the error is the fact that the client is built with ssr: false, but the server is built with ssr: true. I am just guessing though... What is the error you get?

This is the error I get: https://github.com/nuxt/framework/issues/886#issuecomment-1012154169

**Comment by rodrigogs** [comment]
Of course I'm running `nuxi generate` before

**Comment by oysandvik94** [comment]
@blazmrak I'm still getting a similar error on Nuxt3. 

ssr: false,
target: 'static',

```
λ yarn build
yarn run v1.22.17
$ nuxi build
Nuxt CLI v3.0.0-27338323.1e98259                                                                                                                                                                                                    17:02:28
i Vite warmed up in 78ms                                                                                                                                                                                                            17:02:31

 ERROR  [vite:load-fallback] Could not load  (imported by node_modules\nuxt3\dist\app\bootstrap.mjs): ENOENT: no such file or directory, open ''                                                                                    17:02:31


 ERROR  Could not load  (imported by node_modules\nuxt3\dist\app\bootstrap.mjs): ENOENT: no such file or directory, open ''                                                                                                         17:02:31




 ERROR  [!] Error: unfinished hook action(s) on exit:                                                                                                                                                                               17:02:31
(vite:load-fallback) load "frontend/node_modules/vue/dist/vue.runtime.esm-bundler.js"
(vite:load-fallback) load "frontend/node_modules/nuxt3/dist/app/index.mjs"
(vite:load-fallback) load "frontend/node_modules/@vueform/multiselect/themes/default.scss"
(vite:load-fallback) load "frontend/node_modules/nuxt3/dist/app/components/nuxt-root.vue"
(vite:load-fallback) load "frontend/node_modules/nuxt3/dist/pages/runtime/app.vue"


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by oysandvik94** [comment]
@blazmrak I get the same without target: 'static'. 

Also I see in my comment that I used v3.0.0-27338323.1e98259 for that example, but I get the same error with v3.0.0-27367970.e0fec80.

**Comment by rodrigogs** [comment]
> @oysandvik94 and if you remove target: 'static'?
> 
> I think this is a broader issue than just "build fails with SSR off". I think it is dependent on a lot of stuff. First is Bridge/Nuxt3, second is Webpack/Vite, third might be interaction with other parameters. So I think this issue should be split into at least four and if not we should at least specify how we expect the code to behave. @pi0 What is your opinion on this?
> 
> I will try to replicate all four scenarios by the end of the weekend and see if I can come up with anything useful...

If you want to reproduce it, my code is here: https://github.com/rodrigogs/juquinha/blob/refactoring/modules/web/nuxt.config.js#L117

Just remove the `nitro: false` part and try `nuxi generate`

**Comment by blazmrak** [comment]
I have done some testing and it seems that the problem with webpack is the same as was with vite. Nuxt 3 does not build a server when webpack is used unless the [guard](https://github.com/nuxt/framework/blob/33ebb01d7f0c25b5b89b767d9a0749084b0946c4/packages/webpack/src/webpack.ts#L88) is removed. I have not tried Bridge yet, so for you guys who are on Bridge, please do the following:
1. { vite: true, ssr: false } in config
2. edit the node_modues/@nuxt/vite-builder/dist/index.mjs
2.1 Search for ssr: true and change it to ssr: false
3. run nuxi build/generate or whatever

@pi0 do you remember why you put the guard around the generation of the server? The guard was added 21. 7. 2021 in "support ssr: false" feat. #351 which I don't know any details about.

**Comment by farhodhojiev** [comment]
Currently I use bridge version, but none of recommendations work.
Any workaround solution for `bridge` version?

**Comment by rodrigogs** [comment]
> Currently I use bridge version, but none of recommendations work. Any workaround solution for `bridge` version?

Disabling nitro should work

**Comment by rodrigogs** [comment]
> > > Currently I use bridge version, but none of recommendations work. Any workaround solution for `bridge` version?
> > 
> > 
> > Disabling nitro should work
> 
> Yes, it will work, but after can't start `nuxi preview` (nitro.json not found).

I never tried `nuxi preview`, why don't you use `nuxi generate` followed by `nuxi dev`?

**Comment by rodrigogs** [comment]
I updated the version with the fix and a new bug started happening.

```
> nuxi generate

Nuxt CLI v3.0.0-27383750.2b86345                                                                                                                                           10:01:13

 ERROR  Cannot find module 'nuxt'                                                                                                                                          10:01:14
Require stack:
- /home/rodrigosilva/Workspace/juquinha/modules/web/index.js

  Require stack:
  - index.js
  at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
  at Function.resolve (internal/modules/cjs/helpers.js:99:19)
  at Function.p [as resolve] (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/jiti@1.12.9/node_modules/jiti/dist/jiti.js:1:52925)
  at resolveModule (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/@nuxt+kit-edge@3.0.0-27383750.2b86345/node_modules/@nuxt/kit-edge/dist/index.mjs:221:29)
  at importModule (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/@nuxt+kit-edge@3.0.0-27383750.2b86345/node_modules/@nuxt/kit-edge/dist/index.mjs:243:24)
  at loadNuxt (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/@nuxt+kit-edge@3.0.0-27383750.2b86345/node_modules/@nuxt/kit-edge/dist/index.mjs:649:92)
  at async Object.invoke (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/nuxi-edge@3.0.0-27383750.2b86345/node_modules/nuxi-edge/dist/chunks/generate.mjs:35:18)
  at async _main (/home/rodrigosilva/Workspace/juquinha/node_modules/.pnpm/nuxi-edge@3.0.0-27383750.2b86345/node_modules/nuxi-edge/dist/chunks/index.mjs:384:7)
```

**Comment by pi0** [comment]
Hi @rodrigogs. It seems an issue with upgrade process. Can you please remove `node_modules` directory and use `npx nuxi upgrade --force`?

**Comment by rodrigogs** [comment]
> Hi @rodrigogs. It seems an issue with upgrade process. Can you please remove `node_modules` directory and use `npx nuxi upgrade --force`?

I always delete node_modules, package-lock and reinstall everything.

But apparently the recently launched version `3.0.0-27383827.b803fdb` fixes the problem, but I still cannot build it with `nitro`.

```
 ERROR  Rollup error: Could not load /home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs (imported by ../../node_modules/.pnpm/@nuxt+nitro-edge@3.0.0-27383827.b803fdb_acorn@8.7.0/node_modules/@nuxt/nitro-edge/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '/home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs'


 FATAL  Could not load /home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs (imported by ../../node_modules/.pnpm/@nuxt+nitro-edge@3.0.0-27383827.b803fdb_acorn@8.7.0/node_modules/@nuxt/nitro-edge/dist/runtime/app/render.mjs): ENOENT: no such file or directory, open '/home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs'




   ╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                                                                      │
   │   ✖ Nuxt Fatal Error                                                                                                                                 │
   │                                                                                                                                                      │
   │   Error: Could not load /home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs (imported by               │
   │   ../../node_modules/.pnpm/@nuxt+nitro-edge@3.0.0-27383827.b803fdb_acorn@8.7.0/node_modules/@nuxt/nitro-edge/dist/runtime/app/render.mjs): ENOENT:   │
   │   no such file or directory, open '/home/rodrigosilva/Workspace/juquinha/modules/web/node_modules/.cache/nuxt/dist/server/server.mjs'                │
   │                                                                                                                                                      │
   ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

```

**Comment by rodrigogs** [comment]
I'm using this project to test it: https://github.com/rodrigogs/juquinha/tree/master/modules/web

**Comment by Fire7** [comment]
Any updates here? 
I'm trying on `v3.0.0-27394404.1672148`, and build success but there are no entry point in `.output/public/` like index.html
Nuxt3 project.

**Comment by Fire7** [comment]
If anyone has the same problems (build success, but no entry point), here is script that generates `index.html` from build result. Just run it from the root of the project (`node generateIndexHtml.mjs`):

```javascript
// Nuxt 3 not support building SPA for now
// track: https://github.com/nuxt/framework/issues/886
// Create entry point index.html manually
// Must be call from root dir

'use strict';

console.log('Generating index.html...')

import fs from 'fs'
import path from 'path'

const serverSource = fs.readFileSync(path.resolve('.output/server/chunks/nitro/server.mjs'), 'utf-8')
const clientManifest = JSON.parse(fs.readFileSync(path.resolve('.output/public/_nuxt/manifest.json'), 'utf-8'))

const runtimeConfigMatch = serverSource.match(/const _runtimeConfig =(.*)/)

if (!runtimeConfigMatch) {
    console.log('_runtimeConfig not found in server.mjs, nuxt update?')
    process.exit(1)
}

const runtimeConfig = eval(runtimeConfigMatch[1])
const publicConfig = runtimeConfig.public

let css = []
let entryFile = ''

Object.values(clientManifest).forEach(fileConfig => {
    if (fileConfig.isEntry) {
        entryFile = fileConfig.file
    }

    if (fileConfig.css?.length) {
        css = [ ...css, ...fileConfig.css ]
    }
})

console.log({
    entryFile,
    css,
    publicConfig
})

const resultHtml = `<!DOCTYPE html>
<html>

<head>
 ${css.map(cssSrc => `<link rel="stylesheet" href="/_nuxt/${cssSrc}">`).join(`
 `)}
</head>

<body>
    <div id="__nuxt"></div>
    <script>window.__NUXT__={ serverRendered:false, config: ${JSON.stringify(publicConfig)} }</script>
    <script type="module" src="/_nuxt/${entryFile}"></script>
</body>
</html>
`

fs.writeFileSync(path.resolve('.output/public/index.html'), resultHtml)

console.log('Done')
```

**Comment by yeliulee** [comment]
> 

thanks bro, it works for me

**Comment by yeliulee** [comment]
> If anyone has the same problems (build success, but no entry point), here is script that generates `index.html` from build result. Just run it from the root of the project (`node generateIndexHtml.mjs`):
> 
> ```js
> // Nuxt 3 not support building SPA for now
> // track: https://github.com/nuxt/framework/issues/886
> // Create entry point index.html manually
> // Must be call from root dir
> 
> console.log('Generating index.html...')
> 
> import fs from 'fs'
> import path from 'path'
> 
> const serverSource = fs.readFileSync(path.resolve('.output/server/chunks/nitro/server.mjs'), 'utf-8')
> const clientManifest = JSON.parse(fs.readFileSync(path.resolve('.output/public/_nuxt/manifest.json'), 'utf-8'))
> 
> const runtimeConfigMatch = serverSource.match(/const _runtimeConfig =(.*)/)
> 
> if (!runtimeConfigMatch) {
>     console.log('_runtimeConfig not found in server.mjs, nuxt update?')
>     process.exit(1)
> }
> 
> let runtimeConfig = {}
> eval('runtimeConfig = ' + runtimeConfigMatch[1])
> const publicConfig = runtimeConfig.public
> 
> let css = []
> let entryFile = ''
> 
> Object.values(clientManifest).forEach(fileConfig => {
>     if (fileConfig.isEntry) {
>         entryFile = fileConfig.file
>     }
> 
>     if (fileConfig.css?.length) {
>         css = [ ...css, ...fileConfig.css ]
>     }
> })
> 
> console.log({
>     entryFile,
>     css,
>     publicConfig
> })
> 
> const resultHtml = `<!DOCTYPE html>
> <html>
> 
> <head>
>  ${css.map(cssSrc => `<link rel="stylesheet" href="/_nuxt/${cssSrc}">`).join(`
>  `)}
> </head>
> 
> <body>
>     <div id="__nuxt"></div>
>     <script>window.__NUXT__={ serverRendered:false, config: ${JSON.stringify(publicConfig)} }</script>
>     <script type="module" src="/_nuxt/${entryFile}"></script>
> </body>
> </html>
> `
> 
> fs.writeFileSync(path.resolve('.output/public/index.html'), resultHtml)
> 
> console.log('Done')
> ```

thanks bro, it works for me

**Comment by yaquawa** [comment]
I'm using nuxt-bridge, had the same problem.
Currently, my workaround is:

* use `nuxi generate` instead of `nuxi build`
* in `nuxt.config.ts` make sure you have these settings:
```js
{
  bridge: {
    nitro: false,
  },
  ssr: false,
  target: 'static',
}
```

**Comment by vinayakkulkarni** [comment]
Facing the same issue, followed the workaround given by disabling `nitro` only in production build(s)

```js
bridge: {
    nitro: process.env.NODE_ENV !== 'production',
    autoImports: true,
  },
```

```json
"scripts": {
    "dev": "export NODE_OPTIONS=--openssl-legacy-provider; nuxi dev",
    "start": "export NODE_OPTIONS=--openssl-legacy-provider; nuxi preview",
    "generate": "export NODE_OPTIONS=--openssl-legacy-provider; nuxi generate",
},
```

PS. Ignore the `export NODE_OPTIONS` bit, I have to use it cause of Node v17.

Now, Nitro works in `dev`, but while bundling, it's disabled :D

**Comment by katerlouis** [comment]
I mirated to nuxt-bridge aswell and need to proxy my api calls. So I'm forced to build do `target: 'server'` but our global auth middleware obviously doesn't know about stored tokens when called sever-side, so I went `ssr: false` resulting in this error:

<img width="1496" alt="image" src="https://user-images.githubusercontent.com/1983382/158788426-359dedf2-d8d1-462f-9530-01bd7ff8597d.png">

Is it even a thing to ship a server-build with `ssr: false` or is this unintended behavior?

**Comment by zwx00** [comment]
This issue still exists. Disabling nitro for production fixes the issue.

**Comment by mnlj** [comment]
This issue exists in Nuxt 3 as well, can't really disable nitro there. Atleast not with nitro: false in config file.

**Comment by crazymind** [comment]
Because of this bug I needed to disable nitro as well. Now running in to the next issue:
`$fetch is not defined`

I read that:
> Nitro allows 'direct' calling of routes via the globally-available $fetch helper.

Therefore I assume this will cause a chain reaction as $fetch will not be available and other libraries depending on it. I tried to install `ohmyfetch` and `node-fetch` manually – no luck. Also tried to inject $fetch as a plugin – also no luck. Anyone an idea how to fix this?

**Comment by rodrigogs** [comment]
> @crazymind Please try to keep your comments on-topic. They notify a lot of people. File a new issue or discussion for your unrelated question.

I think @crazymind 's comment is totally related to this issue. It seems that disabling nitro is not even a temporary fix to this bug, because you will not be able to use nuxt properly...

**Comment by DreadIsBack** [comment]
> This issue exists in Nuxt 3 as well, can't really disable nitro there. Atleast not with nitro: false in config file.

Confirm. Tested on 3.0.0-27495976.e31c604
And I also use Nuxt3 (without bridge), and seems there is no workaround for this case.
